src/tokenized-line.coffee

```diff
@@ -127,12 +128,13 @@ class TokenizedLine
 
     rightTokens = new Array(@tokens...)
     leftTokens = []
-    leftTextLength = 0
-    while leftTextLength < column
-      if leftTextLength + rightTokens[0].value.length > column
-        rightTokens[0..0] = rightTokens[0].splitAt(column - leftTextLength)
+    leftScreenColumn = 0
+
+    while leftScreenColumn < column
+      if leftScreenColumn + rightTokens[0].screenDelta > column
+        rightTokens[0..0] = rightTokens[0].splitAt(column - leftScreenColumn)
       nextToken = rightTokens.shift()
-      leftTextLength += nextToken.value.length
+      leftScreenColumn += nextToken.screenDelta
```

Comment:
This is a behavior-preserving change. However, instead of relying on text, I find it easier to reason about columns and screen deltas instead.

What do you think?